File: docs/source/example-code/src/main/java/net/corda/docs/java/tutorial/helloworld/IOUState.java
package net.corda.docs.java.tutorial.helloworld;

import net.corda.core.contracts.ContractState;
import net.corda.core.identity.AbstractParty;
import java.util.List;

// DOCSTART 01
// Add these imports:
import com.google.common.collect.ImmutableList;
import net.corda.core.identity.Party;

// Replace TemplateState's definition with:
public class IOUState implements ContractState {
    private final int value;
    private final Party lender;

    public IOUState(int value, Party lender) {
        this.value = value;
        this.lender = lender;
    }

    public int getValue() {
        return value;
    }

    public Party getLender() {
        return lender;
    }

    @Override
    public List<AbstractParty> getParticipants() {
        return ImmutableList.of(lender);
    }
}
// DOCEND 01

File: docs/source/example-code/src/main/kotlin/net/corda/docs/kotlin/tutorial/helloworld/IOUState.kt
@file:Suppress("MemberVisibilityCanBePrivate")

package net.corda.docs.kotlin.tutorial.helloworld

import net.corda.core.contracts.ContractState

// DOCSTART 01
// Add these imports:
import net.corda.core.identity.Party

// Replace TemplateState's definition with:
class IOUState(val value: Int,
               val lender: Party) : ContractState {
    override val participants get() = listOf(lender)
}
// DOCEND 01
